tests/text/helpers.py

```diff
@@ -0,0 +1,538 @@
+        metric_args: dict with additional arguments used for class initialization
+        device: determine which device to run on, either 'cuda' or 'cpu'
+        fragment_kwargs: whether tensors in kwargs should be divided as `preds` and `targets` among processes
+        input_order: Define the ordering for the preds and targets positional arguments.
+        key: The key passed onto the `_assert_allclose` to compare the respective metric from the Dict output against
+            the sk_metric.
+        kwargs_update: Additional keyword arguments that will be passed with preds and
+            targets when running update on the metric.
+    """
+    if not metric_args:
+        metric_args = {}
+
+    metric = partial(metric_functional, **metric_args)
+
+    # Move to device
+    kwargs_update = {k: v.to(device) if isinstance(v, Tensor) else v for k, v in kwargs_update.items()}
+
+    for i in range(NUM_BATCHES):
+        extra_kwargs = {k: v[i] if isinstance(v, Tensor) else v for k, v in kwargs_update.items()}
+        if input_order == INPUT_ORDER.PREDS_FIRST:
+            lightning_result = metric(preds[i], targets[i], **extra_kwargs)
+        elif input_order == INPUT_ORDER.TARGETS_FIRST:
+            lightning_result = metric(targets[i], preds[i], **extra_kwargs)
+
+        extra_kwargs = {
+            k: v.cpu() if isinstance(v, Tensor) else v
+            for k, v in (extra_kwargs if fragment_kwargs else kwargs_update).items()
+        }
+        if input_order == INPUT_ORDER.PREDS_FIRST:
+            sk_result = sk_metric(preds[i], targets[i], **extra_kwargs)
+        elif input_order == INPUT_ORDER.TARGETS_FIRST:
+            sk_result = sk_metric(targets[i], preds[i], **extra_kwargs)
+
+        # assert its the same
+        _assert_allclose(lightning_result, sk_result, atol=atol, key=key)
+
+
+def _assert_half_support(
+    metric_module: Metric,
+    metric_functional: Callable,
+    preds: TEXT_METRIC_INPUT,
+    targets: TEXT_METRIC_INPUT,
+    device: str = "cpu",
+    **kwargs_update,
+):
+    """Test if an metric can be used with half precision tensors.
+
+    Args:
+        metric_module: the metric module to test
+        metric_functional: the metric functional to test
+        preds: torch tensor with predictions
+        targets: torch tensor with targets
+        device: determine device, either "cpu" or "cuda"
+        kwargs_update: Additional keyword arguments that will be passed with preds and
+                targets when running update on the metric.
+    """
+    y_hat = preds[0]
+    y = targets[0]
+    kwargs_update = {
+        k: (v[0].half() if v.is_floating_point() else v[0]).to(device) if isinstance(v, Tensor) else v
+        for k, v in kwargs_update.items()
+    }
+    metric_module = metric_module.to(device)
+    _assert_tensor(metric_module(y_hat, y, **kwargs_update))
+    _assert_tensor(metric_functional(y_hat, y, **kwargs_update))
+
+
+class TextTester:
```

Comment:
what about we do inherence here? then we do not need to have a copy of `setup_class` and `teardown_class`
```suggestion
class TextTester(MetricTester):
```

Reply:
I thought it would cause typing errors, but that didn't happen. So I sub-classed `TextTester` from `MetricTester`.